### ISSUE TITLE
ML self-learning engine — continuous strategy discovery, parallel execution, and automatic live promotion

### ISSUE THREAD

**Issue by GaryOcean428** [GaryOcean428/poloniex-trading-platform#408]
## Feature
Build a continuous ML learning loop that discovers profitable strategies, runs them in parallel, and automatically promotes proven winners to live trading. The system should learn from every trade, adapt to market regime changes, and create new strategy variants without human intervention.

### 1. Strategy Discovery Engine (extends `autonomousStrategyGenerator.js`)
The existing evolutionary algorithm generates strategies via `generateRandomStrategy()` and `crossoverStrategies()`. Enhance it to:
- **Learn from backtest results**: Feed backtest P&L, drawdown, and Sharpe ratio back into the population fitness function
- **Market regime detection**: Classify current market as trending/ranging/volatile using ADX, ATR, and Hurst exponent. Generate strategies suited to the detected regime.
- **Multi-timeframe strategies**: Generate strategies that combine signals across 5m, 15m, 1h, 4h timeframes
- **Novel indicator combinations**: Use genetic programming to evolve new composite indicators from primitives (SMA, EMA, RSI, MACD, Bollinger, Volume, OBV, ATR)

### 2. Parallel Strategy Execution
Currently strategies run sequentially. Enable parallel execution:
- Run top N (e.g., 10) strategies simultaneously during paper trading, each with isolated position tracking
- Allocate virtual capital proportional to backtest Sharpe ratio (Kelly Criterion already exists)
- Track P&L per strategy in real-time via WebSocket
- Kill underperforming strategies (drawdown > 10% or Sharpe < 0 after 50 trades)
- Clone and mutate top performers to fill empty slots

### 3. Continuous Learning Loop
```
loop forever:
  1. Generate new strategy variants (mutation + crossover of best performers)
  2. Backtest on last 30 days of data
  3. If backtest passes (Sharpe > 1.0, WR > 45%, maxDD < 15%):
     → Promote to paper trading
  4. Paper trade for minimum 7 days or 30 trades
  5. If paper trading profitable (Sharpe > 0.8, real P&L > 0):
     → Score confidence via confidenceScoringService
     → If confidence > 60: RECOMMEND for live trading
  6. Store all results in Postgres for the learning loop to reference
  7. Feed performance data back into strategy generation weights
```

### 4. Strategy Performance Database
Create a `strategy_performance` table:
- `(strategy_id, symbol, leverage, timeframe, strategy_type, backtest_sharpe, backtest_wr, paper_sharpe, paper_wr, paper_pnl, live_sharpe, live_pnl, created_at, status)`
- ML queries this table to understand which (symbol, strategy_type, leverage, timeframe) combinations work
- Bayesian optimization over the parameter space guided by historical performance

### 5. Live Trading Promotion
When a strategy is recommended for live:
- Display on Dashboard: "Strategy X ready for live trading — Paper P&L: +$X.XX over Y days, Sharpe: Z"
- Require one-click user confirmation before going live (never auto-trade live without confirmation)
- Start with minimum position size (1% of balance = ~$0.27)
- Scale position size up gradually as live performance confirms paper trading results

## Dependencies
- Depends on: #403 (Fix Python ML), #404 (Unblock paper trading), #405 (Leverage-aware strategies)
- Builds on: `autonomousStrategyGenerator.js`, `backtestingEngine.js`, `paperTradingService.js`, `confidenceScoringService.js`, `fullyAutonomousTrader.ts`

## Files to create/modify
- **New**: `apps/api/src/services/strategyLearningEngine.ts` — Main learning loop
- **New**: `apps/api/src/services/parallelStrategyRunner.ts` — Parallel paper trading executor
- **New**: `apps/api/src/db/migrations/` — Create `strategy_performance` table
- **Modify**: `apps/api/src/services/autonomousStrategyGenerator.js` — Add learning feedback, regime detection
- **Modify**: `apps/api/src/services/paperTradingService.js` — Support multiple concurrent sessions per pair
- **Modify**: `apps/api/src/services/backtestingEngine.js` — Add walk-forward validation (train/test split)

## Acceptance criteria
- [ ] Strategies are continuously generated and backtested without manual intervention
- [ ] Multiple strategies run in parallel during paper trading
- [ ] Underperformers are automatically killed and replaced
- [ ] Top performers are cloned with parameter mutations
- [ ] Performance data stored in Postgres per (symbol, strategy, leverage, timeframe)
- [ ] ML uses historical performance to guide future strategy generation
- [ ] Live trading recommendation appears on Dashboard when a strategy qualifies
- [ ] One-click confirmation required before any live trade execution
- [ ] Walk-forward validation prevents overfitting in backtests

**Comment by GaryOcean428** [comment]
## QIG-Informed Enhancements for the ML Learning Loop

See #410 for the full mapping, but here are the three highest-impact QIG applications specific to this issue:

### 1. Replace Adam with Fisher-Rao Natural Gradient
The evolutionary algorithm in `autonomousStrategyGenerator.js` generates parameter variants. When the ML models in `kernels/core/` train on backtest results, they currently use standard optimizers. Fisher-Rao natural gradient (validated at 1.9–2.2× improvement over Adam, EXP-055) would make model training converge faster with less data — critical when each paper trading cycle only produces ~30 trades before evaluation.

### 2. Principled Multi-Timeframe Weighting (Bridge Law)
The learning loop proposes generating multi-timeframe strategies. The bridge law (τ ∝ J^0.74, R²>0.96, seed-robust) gives a principled weighting: `w(tf) = (base_period / tf)^0.74`. This replaces ad hoc weighting and is backed by 45-point sweeps at two independent seeds.

### 3. Censoring Flag for Paper Trading Evaluation
The `is_censored_tau_dec` pattern should be applied to strategy evaluation. When a paper trading session:
- Hits max drawdown → censored (true loss unknown)
- Hits the 7-day minimum window edge → censored
- Forced position close at session end → censored

Fit performance metrics with and without censored data. One censored outlier destroyed our physics R² from 0.95 to 0.41 — the same will happen to strategy Sharpe ratios if censored sessions pollute the fitness function.

### 4. Regime-Aware Strategy Generation
The strategy generation step (`generateRandomStrategy()`) should be regime-conditioned. The QIG phase diagram shows three distinct regimes. When the learning loop detects a regime change (κ sign flip), it should:
- Kill strategies optimized for the old regime
- Fast-clone strategies suited to the new regime
- NOT average across regimes (this destroys both signal types)

This directly improves the "Market regime detection" item already in the issue body.
